packages/embark/src/bin/embark.js

```diff
@@ -71,11 +78,31 @@ function EmbarkBin(binpath, kind) {
   this.pkgJson = undefined;
 }
 
-EmbarkBin.prototype.exec = function (embarkJson) {
+EmbarkBin.prototype.exec = function (embarkConfig) {
+  if (!embarkConfig) {
+    embarkConfig = findEmbarkConfig();
+    var dappPath = embarkConfig.dirname;
+    process.chdir(dappPath);
+    process.env.DAPP_PATH = dappPath;
+    process.env.PWD = dappPath;
+  }
   var Cmd = require('../cmd/cmd');
-  var cli = new Cmd({embarkConfig: embarkJson.json});
-  if(_logged) { console[this.loglevel](); }
-  cli.process(process.argv);
+  var embarkBin = this;
+  Promise.resolve(embarkConfig.contents)
+    .then(function (embarkConfig) {
+      try {
+        var cli = new Cmd({embarkConfig: embarkConfig});
```

Comment:
Cosmetics, but could probably be just `new Cmd({ embarkConfig })`

Reply:
That short-hand syntax is part of ES6 — I didn't think it was supported in Node v4.0.0, but it is, so I'll make the change, thanks! (eventually the v4 consideration will be tossed aside, but until a larger refactor I'd like to stick to it)

Reply:
Done.